source/agent/streaming/index.js

```diff
@@ -120,25 +147,119 @@ module.exports = function (rpcClient, selfRpcId, parentRpcId, clusterWorkerIP) {
         }
 
         var conn = null;
+        var srtresult = null;
         switch (connectionType) {
             case 'internal':
                 conn = internalConnFactory.fetch(connectionId, 'in');
                 if (conn)
                     conn.connect(options);
                 break;
             case 'streaming':
-                conn = createAVStreamIn(connectionId, options);
-                break;
+		      if(options.connection.type == 'srt') {
+                var minport = global.config.external.minport;
+                var maxport = global.config.external.maxport;
+                var srtmode = 'listener';
+
+                if(options.connection.mode) {
+                    srtmode = options.connection.mode;
+                }
+                if(srtmode == 'listener') {
+                    var portscanner = require('portscanner');
+                    portscanner.findAPortNotInUse([minport,maxport], '0.0.0.0').then(port => {
+                        log.debug('Port is available!', port);
+
+                        var srturl = 'srt://0.0.0.0:'+ port + '?mode=' + srtmode;
+                        if(options.connection.latency) {
+                            srturl = srturl + '&latency=' + options.connection.latency;
+                        }
+
+                        if(options.connection.listen_timeout) {
+                            srturl = srturl + '&listen_timeout=' + options.connection.listen_timeout;
+                        }
+
+                        var srtoptions = {
+                            controller: options.controller,
+                            connection: {
+                                url:srturl,
+                                transportProtocol:'udp'
+			                 },
+                            media: options.media};
+
+                        var networkInterface = global.config.external.network_interface;
+                        var address = getIP(networkInterface);
+
+                        var srtip = global.config.external.ip_address;
+                
+                        if (srtip != "") {
+                            address.push(srtip);
+                        }
+
+                        srtresult = {
+                            ip : address,
+                            port : port,
+                            streamid: connectionId
+                        };
+
+                        log.debug("srt result is:", srtresult);
+			             log.debug('srt options is:', srtoptions);
```

Comment:
Bad indent, you can use some tools like js beautify to format your code before submit.